src/Elm/Parser/Imports.elm

```diff
@@ -3,113 +3,92 @@ module Elm.Parser.Imports exposing (importDefinition)
-
-                                                Nothing ->
-                                                    case maybeExposingList of
-                                                        Just exposingListValue ->
-                                                            let
-                                                                (Node range _) =
-                                                                    exposingListValue.syntax
-                                                            in
-                                                            range
+    ParserFast.map6WithStartLocation
+        (\start commentsAfterImport ((Node modRange _) as mod) commentsAfterModuleName maybeModuleAlias maybeExposingList commentsAfterEverything ->
+            let
+                endRange : Range
+                endRange =
+                    case maybeModuleAlias of
+                        Just moduleAliasValue ->
+                            let
+                                (Node range _) =
+                                    moduleAliasValue.syntax
+                            in
+                            range
 
-                                                        Nothing ->
-                                                            modRange
+                        Nothing ->
+                            case maybeExposingList of
+                                Just exposingListValue ->
+                                    let
+                                        (Node range _) =
+                                            exposingListValue.syntax
                                     in
-                                    { comments =
-                                        commentsAfterImport
-                                            |> Rope.prependTo commentsAfterModuleName
-                                            |> Rope.prependTo
-                                                (case maybeModuleAlias of
-                                                    Nothing ->
-                                                        Rope.empty
+                                    range
 
-                                                    Just moduleAliasValue ->
-                                                        moduleAliasValue.comments
-                                                )
-                                            |> Rope.prependTo
-                                                (case maybeExposingList of
-                                                    Nothing ->
-                                                        Rope.empty
+                                Nothing ->
+                                    modRange
+            in
+            { comments =
+                commentsAfterImport
+                    |> Rope.prependTo commentsAfterModuleName
+                    |> Rope.prependTo
+                        (case maybeModuleAlias of
```

Comment:
Slightly faster would likely be to do
```elm
|> (case maybeModuleAlias of
    Nothing -> identity
    Just moduleAliasValue -> Rope.prependTo moduleAliasValue.comments
   )
```
but I think it would be even faster to introduce a variable:
```elm
let
  commentsUntilAfterModuleName =
      commentsAfterImport
                    |> Rope.prependTo commentsAfterModuleName
in
(case maybeModuleAlias of
    Nothing -> commentsUntilAfterModuleName
    Just moduleAliasValue -> Rope.prependTo moduleAliasValue.comments commentsUntilAfterModuleName
)
  |> Rope.prependTo ...
```

(This pattern is used in a number of places and files, so if one of these is chosen, this can be applied elsewhere as well)

Reply:
went with B as that should be a tiny bit faster indeed (without seeing any perf improvements, but that's expected)